src/app/data-planes/data/index.ts

```diff
@@ -0,0 +1,202 @@
+      return item.health && !item.health.ready
+    })
+    .map((item: {port: any; tags: {[x: string]: any;};}) => `Inbound on port ${item.port} is not ready (kuma.io/service: ${item.tags['kuma.io/service']})`)
+
+  let status: string
+  switch (true) {
+    case dataplane.networking.inbound.length === 0:
+      status = 'online'
+      break
+    // if errors and inbounds are equal, even if they are both 0
+    // then we are offline
+    case errors.length === dataplane.networking.inbound.length:
+      status = 'offline'
+      break
+    // otherwise any errors at all, we are degraded
+    case errors.length > 0:
+      status = 'partially_degraded'
+      break
+    default: {
+      const proxyOnline = dataplaneInsight.subscriptions.some(
+        (item: DiscoverySubscription) => item.connectTime && item.connectTime.length && !item.disconnectTime,
+      )
+      status = proxyOnline ? 'online' : 'offline'
+    }
+  }
+  return status
+}
+function getTag(dataplane: Dataplane, key: string): string {
+  return getTags(dataplane).find((item: { label: string }) => item.label === key)?.value ?? ''
+}
+function getTags(dataplane: Dataplane): Label[] {
+  let tags: [string, string][] = []
+  switch (true) {
+    case dataplane.networking.inbound.length > 0:
+      tags = dataplane.networking.inbound
+        .filter((inbound) => 'tags' in inbound)
+        .flatMap((inbound) => Object.entries(inbound.tags))
+      break
+    case Object.keys(dataplane.networking.gateway?.tags || {}).length > 0:
+    // default:
+      // gateway data plane has no inbounds, but has tags embedded in gateway branch
+      tags = Object.entries(dataplane.networking.gateway?.tags ?? {})
+      break
+  }
+
+  const uniqueTags = Array.from(new Set(tags.map(([key, value]) => `${key}=${value}`)))
+  uniqueTags.sort((tagPairA: string, tagPairB: string) => tagPairA.localeCompare(tagPairB))
+  return uniqueTags
+    .map((tagPair: string) => tagPair.split('='))
+    .map(([label, value]) => ({ label, value }))
+}
+function getVersion(item: DataplaneInsight): Version {
+  return item.subscriptions.length === 0 ? Version.fromJSON({}) : Version.fromJSON(item.subscriptions[item.subscriptions.length - 1].version)
+}
+
+function getDependencies(item: DataplaneInsight): Record<string, BuildVersion> {
+  const version = getVersion(item)
+  const deps: Record<string, BuildVersion> = {}
+  if (version.envoy) {
+    deps.envoy = version.envoy
+  }
+  if (version.kumaDp) {
+    deps.kumaDp = version.kumaDp
+  }
+  if (version.dependencies) {
+    Object.entries(version.dependencies).forEach(([key, value]) => {
+      deps[key] = {
+        version: value,
+      }
+    })
+  }
+  return deps
+}
+
+function getLastUpdated(insight: DataplaneInsight): Date {
+  return new Date(Math.max(
+    ...insight.subscriptions.reduce((prev: number[], item: DiscoverySubscription) => {
+      return prev.concat([new Date(item.status?.lastUpdateTime)?.getTime() ?? 0])
+    }, []),
+  ))
+}
```

Comment:
**Question**: Why are you not using the existing functions for the utility functions inlined here? A lot of these functions are reimplementations of existing functionality.

Can you please use those existing functions? This also makes reviewing this PR much harder.

Reply:
From what I remember, once we don't need all the `?`s `||`s etc etc. The code becomes a lot easier to write and the functions become less complicated and easier to read.

Reply:
I understand that, but couldn’t we do this when we’re ready to replace the functions instead of duplicating them now? This increases the review work as the functions aren’t just exact duplicates but also contain changes themselves.

Reply:
Well half of the reason for this was to show the benefit of doing this. Which is why I added new functions in this file (with the intention of removing the old ones later)

@lahabana could you make a decision on this one please? Should we leave as is in this PR, or should I bring in the old functions here again?

Reply:
Ok so in the end they are not reimplementations because the context is different?

Where is this previous code?

Reply:
Mostly in `src/utilities/dataplane.ts` which from looking at it was the old way of making sure the data reshaping was in a data layer rather than a view layer.

Reply:
>Ok so in the end they are not reimplementations because the context is different?

If I understand this correctly, they are mostly the same apart from the fact that we don't need the additional type checking in there because we do that at a higher level now. Hope that answers this question!

Reply:
Little unfortunate that there are no tests when adding this.
In any case it fells like we need to rewrite them anyway as the typing is different no? It's unfortunate but seems inevitable

Reply:
> Little unfortunate that there are no tests when adding this.

Yes absolutely agree. It would be really good as a follow up at some point to write unit tests across this whole data layer.

> In any case it fells like we need to rewrite them anyway as the typing is different no? It's unfortunate but seems inevitable

Ok so sounds like we leave as is. I'll resolve this one, but please let me know if anyone thinks we should unresolve

Reply:
can you open an issue for adding the tests please?

Reply:
For reference https://github.com/kumahq/kuma-gui/issues/1264